DEVELOP.rst

```diff
@@ -174,9 +174,84 @@ Before creating a new distribution, a new version and tag should be created:
 
 - Update ``CURRENT`` in ``io.crate.Version``
 
-- Add a section for the new version in the ``CHANGES.txt`` file
+- Prepare the release notes
 
-- Commit your changes with a message like "prepare release x.y.z"
+  If a technical writer is available, ask them to complete these steps.
+
+  - Create a new file called ``docs/release_notes/X.Y.Z.txt``
+
+  - The file header should look like this::
+
+        .. _version_X.Y.Z:
+
+        =============
+        Version X.Y.Z
+        =============
+
+        Released on YYYY/MM/DD.
+
+    Be sure to replace ``YYYY/MM/DD`` and both instances of ``X.Y.Z`` with the
+    date of this release and the version number of this release, respectively.
+
+  - Discuss with the engineering team what the minimum version of CrateDB you
+    must be using to upgrade to this version. You will need to know this before
+    moving on to the next step.
+
+  - Add the upgrade notes
+
+    - If the patch version is zero (e.g. 1.1.0), add this note::
+
+          .. NOTE::
+
+             If you are upgrading a cluster, you must be running CrateDB A.B.C or higher
+             before you upgrade to X.Y.Z.
+
+             You cannot perform a :ref:`cluster_upgrade` to this version. Any upgrade to
+             this version will require a full cluster restart.
+
+      Here, replace ``A.B.C`` with the minimum upgrade version. And be sure to
+      replace ``X.Y.Z`` with the version of this release.
+
+    - Else, if the patch version is not zero (e.g. 1.1.1), add this note::
+
+          .. NOTE::
+
+             If you are upgrading a cluster, you must be running CrateDB A.B.C or higher
+             before you upgrade to X.Y.Z.
+
+             If you want to perform a :ref:`cluster_upgrade`, your current CrateDB
+             version number must be :ref:`version_X.Y.0`. Any upgrade from a version
```

Comment:
this is not always true. when the hotfix is on a not stable release it can be possible that it is not possible to do a  rolling upgrade even vor patch versions